notion2md/convertor/block.py

```diff
@@ -22,41 +22,51 @@ def __init__(self, config: Config, client: NotionClient, io: IO = None):
         self._config = config
         self._client = client
         self._io = io
-        self._continued_numbered_list = False
-        self._numbered_list_number = 1
 
     def convert(self, blocks: dict) -> str:
         outcome_blocks: str = ""
         with concurrent.futures.ThreadPoolExecutor() as executor:
             results = executor.map(self.convert_block, blocks)
-            outcome_blocks = "".join([result for result in results])
+            outcome_blocks = "".join([result["text"] for result in results])
```

Comment:
That's an interesting solution. By resetting the number when parsing child blocks, you can ensure that indented blocks retain their sequence. My only concern is the change in the return method, which could potentially throw an unexpected exception somewhere. Have you ever run the application in a CLI environment besides StringExporter? If not, I suggest you try it out. If it runs well, then I can proceed with the merge.